constraint/pkg/client/client_test.go

```diff
@@ -606,3 +606,66 @@ func TestRemoveConstraint(t *testing.T) {
 		})
 	}
 }
+
+func TestAllowedDataFields(t *testing.T) {
+	inventoryTempl := createTemplate(name("fake"), crdNames("Fake"), targets("h1"))
+	inventoryTempl.Spec.Targets[0].Rego = `
+package something
+
+violation[{"msg": "msg"}] {
+	data.inventory = "something_else"
+}
+`
+
+	tc := []struct {
+		Name          string
+		Handler       TargetHandler
+		Template      *templates.ConstraintTemplate
+		ErrorExpected bool
+	}{
+		{
+			Name:          "Inventory Not Used",
+			Handler:       &badHandler{Name: "h1", HasLib: true},
+			Template:      createTemplate(name("fakes"), crdNames("Fakes"), targets("h1")),
+			ErrorExpected: false,
+		},
+		{
+			Name:          "Inventory Used",
+			Handler:       &badHandler{Name: "h1", HasLib: true},
+			Template:      inventoryTempl,
+			ErrorExpected: true,
+		},
```

Comment:
Should we add a test case where `Inventory` used is allowed?

Reply:
Good idea, added.